src/gg.regression.unity.bots/Runtime/Scripts/RGOverlay/RGSequenceEditor.cs

```diff
@@ -168,13 +206,24 @@ public int ClearAvailableSegments()
 
         /**
          * <summary>
-         * Save the current Sequence to disk
+         * Save or update the current Sequence
          * </summary>
          */
         public void SaveSequence()
         {
-            var addedSegments = _dropZone.GetChildren();
+            if (!string.IsNullOrEmpty(_existingSequencePath))
+            {
+                CurrentSequence.segments.Clear();
+
+                // If the Sequence being edited is renamed, we must delete the original Sequence
+                // first. Else we will end up with a duplicate
+                if (CurrentSequence.name != NameInput.text)
+                {
+                    BotSequence.DeleteSequenceAtPath(_existingSequencePath);
```

Comment:
Instead of deleting and recreating... we should at least consider passing the correct path to `BotSequence.SaveSequenceAsJson` instead.  At least when we're editing, we already have the correct path to save to from loading the BotSequence.

I'm not saying we have to do this as it also has downsides in that the filename will become disconnected from the sequence name... but at least ponder its plusses and minuses

Reply:
Good feedback 👍🏻 
I share you concern about having file names and sequence names disconnected. Your point about a crash between the delete and create is valid, I'll make a Linear task to make this process more robust

Reply:
and I'll update to do the create first, delete second

Reply:
After updating the code, I'm not sure if a ticket to improve this is worth it. In the worst case, we save first and the game crashes before we can delete the original. The user will be left with a strange duplicate, but it seems pretty rare. What do you think Zack?

Reply:
that seems fine.. we didn't lose their work, they can manually remove the old un-deleted entry